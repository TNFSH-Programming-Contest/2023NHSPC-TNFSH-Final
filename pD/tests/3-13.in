50 40
191332713 715185045 519715144 377840595 154208624 159959421 254484783 526753520 337232219 880850586 752825753 735383814 332663479 928083869 381070482 422399102 952613892 594056888 681155731 493338632 815261858 162293642 886624283 226969671 -179095733 -209734135 67282192 776019426 603921787 642721627 417886034 -63444650 31657748 116706884 603220475 -136559872 -117312739 241747810 -43322285 -97706193
-170775151 627938920 -238655030 -9712579 -150817011 434912575 -268674532 153284668 850519043 88317196 994769331 943623239 -133212051 346880356 326560320 719390181 90279077 -187311955 257999406 673828425 10805264 748109945 -227438574 696257109 -12410870 923850081 766102566 912554489 745362790 726749473 362184703 516758977 651169569 779482389 172026028 821866976 685595930 -68976320 247461782 341861207
-232755471 847199749 120311678 267644903 -160201763 284872210 16750058 984053279 56826477 -239434823 -186692565 889031126 -228723592 257155716 898597428 482765204 803453554 -87668261 101202385 717404555 227425967 -164407474 142981694 31825149 90416767 362395535 367090192 761838960 6441951 466005543 320609991 -11301273 -169466219 -187929957 536508024 544678729 777602606 435845242 -152098972 325811960
595656281 175019855 -167818852 -186355620 378396531 737775842 -129447258 329152282 392025097 299817429 368746467 420648843 802284567 -133736827 631276893 -12228813 -280033461 -183734097 -79763987 580896408 275302446 701317785 -263355567 866876127 331811699 793508471 -39502677 470102412 276358638 627287687 666224990 616325784 -30651762 928607820 949604737 -207437916 960146038 875958748 861981104 168002587
983213620 496899729 -17599520 882792004 983433674 473053058 503232062 291948394 290007377 825816069 784542244 516594856 784885996 754612894 427043155 224005 300724764 254294900 251065537 724886349 244056228 901411093 -72212934 429770643 225976275 -277007222 823299299 -11520117 661510830 562539788 757482385 -179345207 140158080 281087915 564146546 688331341 106145597 952544428 439661302 514780888
43612981 895360614 482288118 806523143 947104280 692232502 659114620 -158180376 829852627 529627327 -244596772 688472134 -61769941 -12509811 -23122885 690729564 -2997487 20955292 258806107 305688712 396896413 -177127840 579721633 -62100764 251309025 964377190 862407945 485880457 112422414 -89518730 26478954 630819605 -224418955 -127551125 -189406042 627747329 566169275 986616305 143709993 -270404776
556538149 447592646 -45906916 -124190812 903184923 296089705 313446181 256935842 367498397 941920883 377847722 -291540996 809826234 -64955318 245392510 507859015 413575566 360084324 774855291 171927703 619216223 875804015 48616521 767326983 770382228 665851915 532623501 572882728 81007042 -108692924 843569878 667090215 -140450533 77052496 505227942 493134488 765707539 -155765027 798269869 -9683422
672027924 -203255600 250837110 -22958719 977175425 572750557 228575505 967737113 936348178 63368315 -81513218 483408223 506157157 194066235 -17337614 367965455 -190684795 776631111 336683995 162745829 -279136333 65277353 649211882 172015984 49211073 115031789 347644088 266818675 -21182849 -88476204 947669665 226402056 777191411 150784646 358824018 -171556506 731765292 166542278 -160528225 -180001465
-226927526 514908212 290031083 803604041 541883523 464732616 768390679 49054408 845177159 8452851 779876935 -31415523 254762555 771715942 463249432 208717201 -223722006 -41860375 -83146154 511986674 325149624 -51429539 32151039 193413745 294371217 350251796 57692889 -75255881 203700905 -161736552 55443780 451679191 492325672 529111272 379311139 32192130 -51247023 -38230007 768701078 675088469
761867224 325941996 54377575 3352843 -299294224 -261552650 -175293870 780241238 229220917 -148293032 455951155 939222170 438838205 540161116 73381999 89481858 935729946 438877433 527540166 -158196578 -138667950 884938178 372438886 842377507 581833686 95905747 786117063 679204199 -298494534 128179999 353716763 -197464101 47182081 202482749 28556617 -149150795 291722050 119614453 140503833 -12654961
886001593 113655961 109291994 969728721 941215208 41118557 194268490 348137943 653748160 211682736 -613982 115062744 294480936 371202253 -46725175 851874497 932006144 135605131 -68823186 -172597524 -260117039 219853362 48537473 963657938 644364800 881729929 310756062 49454552 17521563 514590129 -297130595 983596608 513751095 206543901 785466027 965433406 189348866 107004123 394691896 759062436
870184159 627997847 872726935 519949650 377408880 295454841 796036320 -289612299 527877274 -261812866 401596860 318431312 124662977 891555074 854305398 232584721 272944882 -17873717 -136763447 -45647395 3767185 -54242256 457790741 322169750 723048894 233875883 619152321 286530920 600187039 127971686 63715920 110806856 468060093 494154789 134192840 596546677 480475882 -10420396 991043921 523164267
154646637 -78437829 938297785 957711867 552671449 329331246 653821682 -264345179 -97953598 31375321 973625512 963269070 -259247517 880444702 139532438 977480982 199944066 -25830269 300762314 818454791 991421794 -108705749 272297102 758957054 352692874 948593202 602274028 786602403 680276575 878052892 745856540 -201014263 972400568 372480795 293531170 608401883 -244554340 598761656 404945117 146058954
309232480 673598199 675279221 425996831 244824726 851714676 32812989 910486634 7896487 465438854 496405405 397707289 322852553 -286928759 -146197106 711263766 -153544224 475995636 99191348 -169602916 943879035 -25603488 564874792 16892621 819123867 887693272 -42558745 673766410 541803808 513705785 894006679 -188465146 196689267 450347047 157474675 848375905 280265822 -132896686 -101498835 887297858
-137436629 586290531 94640852 -194115691 699900409 498251433 868672646 453015164 642383494 -298430470 242656496 -43169108 240971522 733624765 462020865 947016905 862786741 647446391 752056903 403460227 803797861 -31406555 -253436433 310896044 -298602808 -288587470 -117496667 163950115 982982702 127017335 639197448 172144928 990482276 348106426 934115637 243056880 532462524 219755936 557761337 435953992
-238170109 476640108 375605647 607982131 -209002549 755987144 -82046193 650142427 506996071 658517935 534836426 623203729 926583217 324279738 -57301201 305125874 50058665 -272091688 -101633370 -120162829 254585157 851700785 -282741856 163101173 552148530 408704880 -116601398 959146318 140930104 889192 309011364 519501500 -295577603 455441088 660106453 -23907072 -76899837 316262633 924820542 -134051985
-89617560 810986908 -163699926 954016771 -132420438 197052126 366272530 595534752 -208971383 615537172 619284123 26626049 319158465 441031391 777229151 -126893033 -137034371 491559854 -170125397 120262230 -247471003 -227926111 393823258 743058518 294272297 823855245 484567709 941214771 820100078 750911736 671274250 -164908169 590104700 -51557665 265339236 989063814 178280842 357955042 -207963795 887695076
406723313 249932699 829952050 67007310 -46047789 44498579 727394992 -36835383 738464294 877839829 -76896435 -122436141 -224979915 786344205 -33954960 879823850 337354976 442740543 540537424 30089398 -178387493 889381566 479799549 -275441654 -91462916 972169196 784717440 -264341838 611355127 -21350700 -42209807 987577510 393731264 122761220 523532592 387314185 524504145 279109706 738813416 -169840569
-137085014 787129469 455660017 -259604841 240073239 707276666 715834202 896254462 289097173 45228314 -7684222 811521877 559698391 989055906 -96192910 97923056 608980823 -15927650 -66265085 939522817 909236432 3562373 8843130 752152166 -53073940 -200171174 798622491 953950915 877732296 575476036 44848561 823710355 437458351 317648880 594169503 778092476 160241128 608003234 336193579 473430392
20822931 -253279275 905819262 588343735 705129762 298249106 173395681 142956697 -104581723 -66890445 890888706 -181124008 418896864 578146011 -267563059 943770657 887485849 828756495 -44647825 817885251 594031845 299616086 -151986271 860264803 15746014 834265218 813225285 95895100 353720285 931721394 733932108 189038913 529095082 170102583 385986583 164107735 360793397 979462266 443748279 988812781
-49623671 595837485 929755377 816741510 419318066 -21492373 -25078675 197493871 39469791 480606958 -246006989 337994560 278110008 185954281 650946923 208014058 804901757 686077452 237625775 168824093 849751709 882406195 544516972 -208565729 384423498 298011261 353194405 3608635 979634948 440441886 38772787 666714060 817320091 555298216 228573958 -172391525 881656849 647340200 -26103843 323518486
430416774 524477589 313942422 -151545536 -286031093 770954087 641740760 154065271 228472561 832407019 894904126 275456281 836016424 313369880 329811130 -276105693 173442389 -149120756 -271649240 766742153 825991438 -241700146 167673338 -200034273 830838940 234260088 -31241926 101697201 -109562206 -254440622 632510129 82312916 -154848536 -183321138 724669009 744352338 -87407201 545111886 17245730 570608064
799450775 12338129 -225078988 743879293 595367936 285138113 735580260 91254352 409885947 -78287206 936459748 931293100 735807859 228938555 210591736 -18495631 656521339 431474768 827123532 -95681198 -68860337 276675411 669204934 893705012 363869206 -146814390 117141005 826551976 -97343 927715564 450415991 430649234 81843120 927943090 859130972 608100053 147059473 662582457 911864779 341644958
239772309 540554319 609206205 604652723 571172597 -235867591 -157641843 891196034 -196078582 797301097 37050163 587687636 628361017 -210132452 59806863 894909077 574051954 545481736 468110218 -119982845 -287894469 112177216 381541482 491534713 95271135 602621602 968591823 127097291 964974082 518682953 853377867 830622129 844460506 893409635 911290803 293529673 567446695 742963803 406056380 -253270921
-171188293 412727474 492649736 93182120 446876277 932405508 818985180 452389039 610317538 36536384 487638304 -194028681 -34853234 687960992 32873509 447159348 -220820246 405587436 685448204 -197403158 -202882049 -70214294 256488472 -159498242 -252018877 -26487501 -104783310 559061427 809694488 136078956 -5744416 -212169991 418728949 406885833 27665483 -64902503 696336768 -144882982 646104903 776984249
-200964496 858910964 321616342 148019162 893861162 -87045764 252577069 -239702001 730137660 -182119994 747974357 -173443975 785835934 992835118 567538039 -18023952 103477192 746902898 955674128 338344897 23835722 274972068 55029067 -174990393 287308122 444375070 982536158 495234369 505152443 172481557 413099930 748477624 581203363 662429203 212388579 806568630 923164185 864209545 867089052 11793411
848444070 225026335 505386334 856309528 579768497 531807105 912785081 880501609 487583441 -147499301 875890176 381946066 -50574485 129207834 110672225 -94205353 502593577 -254831849 105437206 310055697 -97059238 165458327 -232057163 943180005 523597850 235562657 722759489 787094131 -290403729 481501696 -183366385 170373495 -266843365 -285122660 459561182 448830562 -152840365 922749521 438284962 796146427
-60930858 -175191897 815231485 808759269 -261521248 -167688504 445983570 146501015 -275479280 593694843 276690020 361871107 206348735 883962981 860879380 900634858 280012551 190581868 14491110 327892290 950764894 455938223 445831240 209385726 685932907 -66834050 277712596 704434375 -60256011 993171866 -61793996 712371101 375987775 160875670 873291674 -120008091 461085861 -217745379 859818537 820717573
834473281 806793278 327676704 -85825755 819485920 607694946 599089160 624657055 634616219 -182986978 719861709 830225135 628381418 231626225 -38151395 388690641 783423727 303349115 7182197 317901894 -91169771 49478841 981757851 307695454 167019777 16102235 509182920 -147604719 211527147 209235276 310715696 513220914 365857380 90726618 -82548699 864712139 -282374864 79920620 121482828 311676197
620425068 4657624 -167650141 748725566 450921325 285981896 -167948853 -60428474 668902588 985187828 -275656165 379664348 816111002 -231448119 964388497 827124406 -93896689 428682052 133426447 499036204 126347119 623961839 -53801597 -53289918 652025270 665604633 644475147 423587872 198936860 497114903 805811156 -255896886 260514384 -270250940 -188206414 229212819 -100699364 -45832267 624551150 -230984029
942150281 302014984 457831812 -130443668 -56668423 967701598 76356352 172563381 -219108979 102699723 244981808 290330954 578010278 -201861849 375575355 942875365 -292342072 560515304 405626303 -55675786 611719928 645137528 575151040 347351496 529009714 929545484 -118893143 -203677535 876333219 -258033934 614025444 861734493 348444678 151763803 664976485 -139088194 924723186 -295942507 522102561 390206097
955053076 789812020 720744232 22735608 -129412626 -35433028 -3422714 356176514 -169469366 526006096 954721843 953403608 938342980 330246483 273079337 889619131 -522111 896787413 999478109 499130054 595036616 67979339 474219033 -273039030 466298683 -23627760 -38034532 326011039 519703713 566772567 311839887 293785786 634340782 -294328516 -265410218 916642466 155552826 566328656 66172103 -151865229
-70181602 594986315 870210667 639577565 864475816 -241195668 727639980 16521477 288291050 792159140 107403008 785693790 231493602 583521770 -108811359 -46451559 -141911373 940740268 623296070 247959409 387065268 -11128426 92372488 455367226 771277696 23438039 684605325 -97448500 555766954 708185997 759765585 463494078 230365202 539749466 -214353757 -68795313 871467783 530559023 209026550 -31142974
-73447263 54708249 980887755 26918884 354419374 542376306 -105968183 758780339 -219055048 724522353 89030192 284269608 115784749 -242560244 806231328 364937134 153993631 427552871 349619868 -162802664 -102246453 178478048 -131191481 -38770475 494081825 244187165 769702082 -147904510 302557228 -114239175 676194040 524284376 734904292 563498491 225155072 308209154 -8779059 903713263 149653496 165990254
55611162 793412461 247027309 785265707 327150418 -2578430 -19013845 364343861 -166942599 115686979 983981053 831500302 418709082 473732961 362772880 -15434357 470373239 -6981459 -170776303 -172964147 182670658 731032523 721590011 363021535 427904426 894952128 646202863 -213968719 630279462 453531013 81665315 635881917 208148114 503887945 -42062934 790883505 636878942 534147282 -149734034 -10714003
772824649 143831691 201507262 -110808013 21661215 57283317 404989511 143039810 891373207 952933951 136795679 274982132 987025329 948617828 305307799 571900904 -17961458 150902330 906284719 746866504 -81301566 -164239935 47414488 381881567 247229914 421799719 548990230 -248668168 80641129 370445153 -79945840 150667758 726315040 946613664 933160265 78296299 993925447 -237990855 65951077 810952406
744895637 547861873 846745082 92196544 -96059847 -129950395 647998044 -183134342 22496517 520350867 608666430 741004479 342826736 140867934 69341675 -81297233 393771997 158804955 -77599393 851313673 8003681 886161354 423037667 919352915 403040256 489675336 862816956 580573697 -133949807 946041877 67445722 21888786 600975654 -226805137 -135401392 -197323048 -259321949 -86416268 781748482 -154357303
388160116 454060987 715620307 69544508 -106939522 749804118 734644347 440459025 -270204854 744799094 171544056 848178720 71479847 737413778 807426076 379470644 -12588815 736912767 262553945 146768520 91640261 -247257967 578258224 774077036 972880645 763877033 -124591349 -21221554 -1420495 479244271 735569625 327156780 25917150 -277089119 -44923114 535680661 -237785737 949507491 429410130 -98700259
426289202 -138673366 981552222 343806624 852295889 824533679 13306490 281098318 -91176858 1474053 -43531175 28451445 -265759072 489691829 550495478 -281751236 697110829 325381972 -73702201 199561825 331489211 228640326 179749288 831433077 639194113 -238327048 -48182894 128927528 947562792 -100242416 149974051 116293230 824037697 905712105 283113636 185162844 571447456 452822200 553408044 451048460
546729312 -50527752 -241642098 930244904 -239640742 575696064 218063125 709537363 600442711 444922505 387484782 248398349 953589544 678765084 -7546424 17191669 438222098 444229765 -163729288 988481558 266949806 966283768 380063638 425089655 94427901 778312693 505515861 153783678 303772174 903649819 338401789 922781136 589278516 108847217 842580977 -260727603 -137423159 -212925190 -289532935 28803376
582951558 -55047413 451309134 819498595 750275610 743697714 167359886 208728199 -297960907 589102467 912812284 620911391 172770015 -180176426 -285002952 -165824077 252482511 28146851 428053445 153627130 -68475087 -132207896 79670941 430217894 865562998 680997274 859215099 -59719927 255593009 657630960 593562549 96281901 228202511 101174135 681809691 317145664 -96766456 210282388 47804423 -279005446
256559009 483229436 616986438 635716094 706328896 701931791 -269472867 803691363 -189756159 447955110 -164536799 692885664 575423391 -261601014 -248667717 954627176 711682038 136443185 57885999 642446471 712288985 497386227 996060670 270655805 -266940012 -175665115 640541259 40218321 143944979 818327464 131393302 -75596705 892661636 79057754 685814657 804905618 524147256 -118726543 850297748 317349816
-227549839 -47089076 817130204 30329281 681059256 267236818 56284895 260957142 -4572019 859159714 220277198 96514891 102043570 -230468133 527999929 -244476291 550332275 90311452 793770118 768965647 -223491898 633262602 -292828040 137454867 423267048 -138148149 947227540 216543148 894075266 -217670886 16588085 -243449668 512061518 207123865 -135340390 243538098 -37597759 45334993 365214726 742710285
4491488 15065288 681568293 79713029 723639479 892055762 814185702 275456376 396575195 439644872 771483467 692049058 873278691 323316044 -122408378 538082524 848212494 523768148 303323039 782123599 835066672 -85868877 895748045 -117488617 548937673 420334587 243415464 657754160 655239534 941354414 450818616 590347398 274687709 -231313657 -24979714 805213062 657048676 702370481 -279419488 705370996
656868187 14667167 -299115934 -159513762 300330730 91046272 683109626 369613290 158878396 78197054 442186780 -8119667 980457014 -164037764 341270280 613513985 542498233 942100443 -116704561 22661488 884214998 959846045 383907644 313527687 -145127108 784100466 -199034523 894482199 758223768 496044223 64973106 -86137711 404940620 -77820520 989181180 962491183 -224252828 37185545 -233214690 589981401
410983949 611831890 268455466 810087440 840125903 524070967 651181109 -181134298 592899173 -159000802 -152752266 757607533 338027833 307176059 404870900 52524503 947493211 -296267787 259563000 103962225 748615951 189418074 916330284 769942301 636515650 137878726 953545078 529997352 747393055 428893065 126054682 444088786 735627860 676622822 457766610 -41584010 736108040 984921756 779785793 274002140
516201703 496175073 -95555969 -203577398 -126177913 27260424 -147138926 157700125 811888126 -282392539 671744251 255665831 267201080 160347748 -17550851 -45563737 -24661889 975696898 876613111 10475821 988828341 197691857 39840109 591148686 365679355 249518057 -284874952 -272665866 137581153 13855726 -248334647 802002542 353492881 158558333 536775436 420182763 610258907 -239553467 934236328 803314630
981550579 504280698 -55960429 405746280 952954112 -131944478 37222576 523685258 103627 -113804687 709652210 36736916 -140656648 -265732484 478570779 59801299 354355298 -19112211 -248752125 643119110 914214706 976898481 790777207 35981222 -111944867 495734934 484358239 -145879289 589633580 220509153 789232066 -260173663 169823491 286755616 113169274 887662815 74104770 262676036 870312406 123335219
257912355 687406640 279884596 320502585 777729974 14072281 -166696811 407282122 192927812 532888428 98904925 856733365 648141005 215579015 92697763 -161402555 91973044 65378234 229013025 103521105 562646758 -287729302 268703851 -100600358 85531492 412538468 762579432 -6791119 -152148258 -38668196 927119897 -164933423 463394982 -219279608 -293891167 -235026644 469612109 934889104 962709548 -22304603
423158464 819967841 -138403499 307568237 444402685 651890249 430952663 -9143481 836259374 877473990 747935571 927614611 308767272 -299075098 997530841 671986956 755650437 605907987 -15747651 568508269 187848903 10263315 -188463903 42831659 -110221588 -4941289 708911536 856394979 178703063 -76177163 -84955768 -211915019 471843003 934388693 147385127 625407395 471738025 726436612 766363218 500172222
