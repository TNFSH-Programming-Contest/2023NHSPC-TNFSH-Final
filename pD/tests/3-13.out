1722383172432
